index_documents_test.go

```diff
@@ -1525,30 +1523,66 @@ func TestIndex_GetDocuments(t *testing.T) {
 			},
 		},
 		{
-			name: "TestIndexGetDocumentsWithLimit",
+			name: "TestIndexGetDocumentsWithFields",
 			args: args{
-				UID:    "TestIndexGetDocumentsWithNoExistingDocument",
 				client: defaultClient,
 				request: &DocumentsQuery{
 					Fields: []string{"title"},
 				},
 				resp: &DocumentsResult{},
 			},
 		},
+		{
+			name: "TestIndexGetDocumentsWithFilter",
+			args: args{
+				client: defaultClient,
+				request: &DocumentsQuery{
+					Filter: []string{"tag = Tragedy"},
+				},
+				resp: &DocumentsResult{},
+				filter: []string{
+					"tag",
+				},
+			},
+			result: 3,
+		},
+		{
+			name: "TestIndexGetDocumentsWithMultipleFilter",
+			args: args{
+				client: defaultClient,
+				request: &DocumentsQuery{
+					Filter: []string{"tag = Tragedy", "book_id = 742"},
```

Comment:
Will this be translated to "tag = Tragedy OR book_id = 742"?

Reply:
`[]string{"tag = Tragedy", "book_id = 742"}` =  `"tag = Tragedy AND book_id = 742"`

```[]interface{}{[]string{"tag = Tragedy", "book_id = 123"}}``` =  `"tag = Tragedy OR book_id = 742"`

So I modify the type of filter `filter` from `[]string` to `interface{}` to allow the user to use both `string` as `[]string` as `interface{}`

Reply:
Sorry, but I didn't get it:

How this: `[]interface{}{[]string{"tag = Tragedy", "book_id = 123"}}` can be different than this `[]string{"tag = Tragedy", "book_id = 742"}`?

To me they should be different arrays: `[]interface{}{[]string{"tag = Tragedy", "book_id = 742"},[]string{...}` no?

Reply:
This is not me; this is from the documentation guide about [filtering with array](https://www.meilisearch.com/docs/learn/advanced/filtering#creating-filter-expressions-with-arrays):
AND `["genres = horror", "director = 'Jordan Peele'"]`
OR `[["genres = horror", "genres = comedy"]]`

But in Golang you can do `[]string{[]string}` if you want to have an array of `[]string`, you should use `interface{}` or maybe there is another way but I don't know it.
But yes `[]interface{}{[]string{"tag = Tragedy", "book_id = 742"},[]string{...}` this should be also doable.